mscore/preferences.cpp

```diff
@@ -38,6 +38,38 @@ void Preferences::init(bool storeInMemoryOnly)
             _settings = new QSettings();
             }
 
+#if defined(Q_OS_WIN)
+      // Use the correct default GUI font on Windows. Qt 5.x and below use the deprecated function GetStockObject() with
+      // DEFAULT_GUI_FONT, which returns MS Shell Dlg 2 in 8 pt. MS Shell Dlg 2 is a virtual font that maps to Tahoma, which has
+      // not been the default Windows GUI font since 2006.
+      //
+      // The correct way to determine the default GUI font is to call SystemParametersInfoW() with SPI_GETNONCLIENTMETRICS and use
+      // the returned lfMessageFont structure. On all versions of Windows from Windows Vista through Windows 10, this typically
+      // returns Segoe UI in 9 pt.
+      //
+      // This problem is slated to be fixed in Qt 6, and the code here is equivalent to the code that's already been written for
+      // Qt 6 in QWindowsFontDatabase::systemDefaultFont(). For details, see: https://bugreports.qt.io/browse/QTBUG-58610
+      NONCLIENTMETRICSW ncm { sizeof(NONCLIENTMETRICSW) };
+      if (SystemParametersInfoW(SPI_GETNONCLIENTMETRICS, ncm.cbSize, &ncm, 0U)) {
+            const auto& font = ncm.lfMessageFont;
+            std::unique_ptr<std::remove_pointer<HDC>::type, void (*)(HDC)> dc(GetDC(NULL), [](HDC dc) { ReleaseDC(NULL, dc); });
+            const auto dpi = dc ? GetDeviceCaps(dc.get(), LOGPIXELSY) : 96;
+            const auto size = static_cast<int>(-font.lfHeight * 72 / dpi);
```

Comment:
Do we need a magic numbers here (96 and 72) rather than one of the macros? Like `PPI`?

Reply:
These are stock conversion formulas that most seasoned Windows developers should recognize instantly. Yes, they are magic numbers, and magic numbers are bad, but there is context here, and each number is used only once.

If you really think I should create named constants for them I will, but I don't feel strongly either way.

Reply:
Nah. Only, maybe, use the existing one(s) (like `PPI` for 72.0)

Reply:
I went ahead and got rid of the magic numbers. The existing constants were both the wrong type, though, so I had to add casts.